code-challenges.test.js

```diff
@@ -52,5 +105,54 @@ const numbersToAdd2 = [0, 7, -8, 12]
 const numbersToAdd3 = []
 // Expected output: []
 
+// describe("arraySum", () => {
+//   it("returns an array, where each number is the accumulating sum of the previous value", () => {
+//     expect(arraySum("array")).toEqual("array of accumulating sums")
+//   })
+// })
+
 
 // b) Create the function that makes the test pass.
+
+// Pseudo Code:
+// declare a function arraySum
+// set the parameter to an array
+// itterate through array
+// declare empty array newArray
+// return value of index 0 as is with conditional
+// from index 1 onwards, add previous value of newArray and move onto next
+// return newArray of values each containing accumulating sum
+
+const arraySum = (array) => {
```

Comment:
mission accomplished! Let's see if hof's can do some heavy lifting for you.

Reply:
I think I remember using for loop at first for this cause i wasn't sure how to define index in the hof map, but I reviewed it later, and the map version of the code should be commented on line 138. Will keep in mind to use hof's more though.